crates/wasmtime/src/component/mod.rs

```diff
@@ -24,7 +24,11 @@ pub use self::instance::{ExportInstance, Exports, Instance, InstancePre};
 pub use self::linker::{Linker, LinkerInstance, ResourceImportIndex};
 pub use self::resource_table::{ResourceTable, ResourceTableError};
 pub use self::resources::{Resource, ResourceAny};
-pub use self::types::{ResourceType, Type};
+pub use self::types::{
+    ComponentFunc as ComponentFuncType, ComponentInstance as ComponentInstanceType,
+    ComponentItem as ComponentItemType, CoreFunc as CoreFuncType, Module as ModuleType,
+    ResourceType, Type,
```

Comment:
Since `types` is a public module it's ok to avoid these reexports to avoid the same type being known under two different names.